public/app/plugins/panel/datagrid/DataGridPanel.tsx

```diff
@@ -162,14 +183,19 @@ export function DataGridPanel({ options, data, id, fieldConfig, width, height }:
     });
   };
 
-  const onColumnMove = (from: number, to: number) => {
+  const onColumnMove = async (from: number, to: number) => {
     const fields = frame.fields.map((f) => f);
     const field = fields[from];
     fields.splice(from, 1);
     fields.splice(to, 0, field);
 
-    dispatch({ type: DatagridActionType.columnMove, payload: { from, to } });
-    publishSnapshot({ ...frame, fields }, id);
+    if (onUpdateData) {
+      const hasUpdated = await onUpdateData([{ ...frame, fields }]);
+
+      if (hasUpdated) {
+        dispatch({ type: DatagridActionType.columnMove, payload: { from, to } });
```

Comment:
what does this do? Won't the panel get new data with the move?

Reply:
This is a special case because of the column width. Since width is not a property on the dataframe, but part of the datagrid column themselves, we also dispatch the `columnMove` to update the columns width. The panel gets the new data with the move, because of the `onUpdateHook` and with the dispatch it also carries the width on the correct position after the column is moved.